### PR TITLE
Histogram MaxAge is configurable and change default quantiles and only collect p90

### PR DESCRIPTION
**Pull request by lfitchett**
Max age can now be configured with 'MetricsHistogramMaxAge'.

Histogram quantiles are now 0.1, 0.5, 0.9 and 0.99

Only p90 value is uploaded as part of rqt

### CODE REVIEW

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Diagnostics/MetricsWorker.cs [hunk 1]
```diff
@@ -40,7 +40,12 @@ public MetricsWorker(IMetricsScraper scraper, IMetricsStorage storage, IMetricsP
             this.uploader = Preconditions.CheckNotNull(uploader, nameof(uploader));
 
             this.metricFilter = new MetricTransformer()
-                .AddAllowedTags(new KeyValuePair<string, string>(MetricsConstants.MsTelemetry, true.ToString()))
+                .AddAllowedTags((MetricsConstants.MsTelemetry, true.ToString()))
+                .AddDisallowedTags(
```
Comment (damonbarry) [inline edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Diagnostics/MetricsWorker.cs:44]: Is this how we make sure RQT only captures P90?

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Diagnostics/MetricsWorker.cs [hunk 2]
```diff
@@ -40,7 +40,12 @@ public MetricsWorker(IMetricsScraper scraper, IMetricsStorage storage, IMetricsP
...
+                    ("quantile", "0.1"),
+                    ("quantile", "0.25"),
```
Comment (damonbarry) [inline edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Diagnostics/MetricsWorker.cs:46]: `0.25` isn't one of the quantiles listed in `MetricsHistogram` below...do we need to disallow it?

**Comment by damonbarry** [comment]
Also...no tests?

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Diagnostics/MetricsWorker.cs (hunk 1 above)
Comment (lfitchett) [reply edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Diagnostics/MetricsWorker.cs:44]: yep

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Diagnostics/MetricsWorker.cs (hunk 2 above)
Comment (lfitchett) [reply edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Diagnostics/MetricsWorker.cs:46]: nope, thats leftover from a previous iteration, removing it

**Review by damonbarry — APPROVED** [review]

**Pull request merged**